src/beta/resources/referral.js

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable no-dupe-class-members,no-unused-vars,class-methods-use-this */
+import superagent from 'superagent';
+
+import T from 'proptypes';
+import base from '../../resources/base';
+import { propTypes as userPropTypes } from '../../resources/user';
+import RequestError from '../../errors/request';
+import API from '../../easypost';
+
+export const propTypes = Object.assign({}, userPropTypes, {
+  name: T.string.isRequired, // mark as required
+  email: T.string.isRequired, // mark as required
+  phone: T.string.isRequired, // handle discrepancy between phone (create Referral) and phone_number (all User functions, retrieve Referral)
+});
+
+export default (api) =>
+  class Referral extends base(api) {
+    static _name = 'Referral';
+
+    static _url = 'referral_customers';
+
+    static key = 'user';
+
+    static propTypes = propTypes;
+
+    static #stripeCreditCardUrl = 'https://api.stripe.com/v1/tokens';
+
+    static #getReferralApi(referralApiKey) {
+      return API.copyApi(api, {
+        apiKey: referralApiKey,
+      });
+    }
+
+    async save() {
+      if (this.phone_number !== undefined) {
+        // handle discrepancy between phone (create Referral) and phone_number (all User functions, retrieve Referral)
+        this.phone = this.phone_number;
+      }
+      return super.save();
+    }
+
+    /**
+     * Update the referral's email address.
+     * @param {string} referralUserId - The referral user's ID.
+     * @param {string} email - The new email address.
+     * @returns {Promise<boolean>}
+     */
+    static async updateEmail(referralUserId, email) {
+      try {
+        const newParams = { user: { email } };
+        await api.put(`${this._url}/${referralUserId}`, newParams);
+        return true;
+      } catch (e) {
+        return Promise.reject(e);
+      }
+    }
+
+    /**
+     * Add a credit card to the referral's account
+     * @param {string} referralApiKey - The referral user's production API key.
+     * @param {object} creditCardDetails  - Credit card details:
+     * @param {string} creditCardDetails.number  - Credit card number
+     * @param {string} creditCardDetails.expirationMonth  - Credit card expiration month
+     * @param {string} creditCardDetails.expirationYear  - Credit card expiration year
+     * @param {string} creditCardDetails.cvc  - Credit card CVC
+     * @param {string} primaryOrSecondary  - Whether to add the card as the primary or secondary card (defaults to primary)
```

Comment:
```suggestion
     * @param {object} creditCardDetails - Credit card details:
     * @param {string} creditCardDetails.number - Credit card number
     * @param {string} creditCardDetails.expirationMonth - Credit card expiration month
     * @param {string} creditCardDetails.expirationYear - Credit card expiration year
     * @param {string} creditCardDetails.cvc - Credit card CVC
     * @param {string} primaryOrSecondary - Whether to add the card as the primary or secondary card (defaults to primary)
```

Should this be single spaced